server/publications/collections/product.js

```diff
@@ -66,48 +66,36 @@ Meteor.publish("Product", function (productId) {
     return this.ready();
   }
   let _id;
+  const selector = {
+    isDeleted: { $in: [null, false] }
+  };
   let productShopId;
   const shop = Reaction.getCurrentShop();
   // verify that parent shop is ready
   if (typeof shop !== "object") {
     return this.ready();
   }
 
-  // selector should come first as default, alterations take place later depending on role
-  const selector = {
-    isVisible: true,
```

Comment:
Not sure I understand why we're removing the requirement that a product be visible by default. Seems safer to do it that way and remove that requirement based role than vice-versa.